### PR TITLE
[KIECLOUD-480] - Allowing value flag as "optional" for LdapExtended login module

### PR DESCRIPTION
**Pull request by msmagnanijr**
See: https://issues.redhat.com/browse/KIECLOUD-480

Signed-off-by: Mauricio Magnani <msmagnanijr@gmail.com>

Thanks for submitting your Pull Request!

Please make sure your PR meets the following requirements:

- [x] Pull Request title is properly formatted: `[RHPAM-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Attached commits represent units of work and are properly formatted
- [x] You have read and agreed to the Developer Certificate of Origin (DCO) (see `CONTRIBUTING.md`)
- [x] Every commit contains `Signed-off-by: Your Name <yourname@example.com>` - use `git commit -s`

### CODE REVIEW

**Comment by spolti** [comment]
can you please check the check buttons that matches the work you've done on this PR?
Thanks.

**Review by spolti — DISMISSED** [review]

**Comment by spolti** [comment]
@jakubschwan @Kaitou786 @desmax74 can you please review?

File: templates/rhpam710-authoring.yaml [hunk 1]
```diff
@@ -323,6 +323,11 @@ parameters:
   name: AUTH_LDAP_BIND_CREDENTIAL
   example: "Password"
   required: false
+- displayName: LDAP Flag Login Module
+  name: AUTH_LDAP_LOGIN_MODULE
+  description: "A flag to set login module to optional. The default value is required"
+  required: false
+  value: "optional"
```
Comment (Kaitou786) [inline templates/rhpam710-authoring.yaml:330]: Shouldn't this `example` instead of `value`?

**Review by Kaitou786 — CHANGES_REQUESTED** [review]

File: templates/rhpam710-authoring.yaml (hunk 1 above)
Comment (jakubschwan) [reply templates/rhpam710-authoring.yaml:330]: Great catch @Kaitou786 . Yes there should be used example. Right now we are overriding default value `required`.
@msmagnanijr please fix this env var in all places

**Review by jakubschwan — APPROVED** [review]
Looks good for me. Only one change is needed and already pointed out, rename of vlaue to example for AUTH_LDAP_LOGIN_MODULE property. After this change can be merged.

File: templates/rhpam710-authoring.yaml (hunk 1 above)
Comment (msmagnanijr) [reply templates/rhpam710-authoring.yaml:330]: @Kaitou786 good catch, tks!

File: templates/rhpam710-trial-ephemeral.yaml [hunk 2]
```diff
@@ -236,6 +236,11 @@ parameters:
   name: AUTH_LDAP_BIND_CREDENTIAL
   example: "Password"
   required: false
+- displayName: LDAP Flag Login Module
+  name: AUTH_LDAP_LOGIN_MODULE
+  description: "A flag to set login module to optional. The default value is required"
+  required: false
+  value: "optional"
```
Comment (Kaitou786) [inline templates/rhpam710-trial-ephemeral.yaml:243]: @msmagnanijr  missed this one :p

**Review by Kaitou786 — CHANGES_REQUESTED** [review]

File: templates/rhpam710-trial-ephemeral.yaml (hunk 2 above)
Comment (msmagnanijr) [reply templates/rhpam710-trial-ephemeral.yaml:243]: thanks :D

**Review by Kaitou786 — APPROVED** [review]

**Review by spolti — APPROVED** [review]

**Pull request merged**